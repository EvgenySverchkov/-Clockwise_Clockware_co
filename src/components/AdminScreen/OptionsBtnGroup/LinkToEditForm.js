import React from 'react';
import {Link, useLocation} from "react-router-dom";

export default function LinkToEditForm(props){
  let location = useLocation();

  function chooseRoute(currentPath){
    let arr = currentPath.split('/');
    switch(arr[arr.length-1]){
      case "clientsList":
        return "/admin/editClient/";
      case "mastersList":
        return "/admin/editMaster/";
      case "townsList":
        return "/admin/editTown/";
      default:
        return currentPath;
    }
  }
  return (
    <Link to={chooseRoute(location.pathname) + props.id}
          className="dropdown-item">Edit</Link>
  );
}
