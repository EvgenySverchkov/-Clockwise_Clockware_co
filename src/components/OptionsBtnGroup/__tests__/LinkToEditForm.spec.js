import React from "react";
import {create} from "react-test-renderer";
import {MemoryRouter} from "react-router-dom";

import LinkToEditForm from "../LinkToEditForm";

describe("Test of <LinkToEditForm />", ()=>{
    const mockProps = {
        id: 1
    }
    it("View when current path is 'admin/clientsList'", ()=>{
        const component = create(
            <MemoryRouter initialEntries={["admin/clientsList"]}>
                <LinkToEditForm {...mockProps}/>
            </MemoryRouter>
        );
        expect(component.toJSON()).toMatchSnapshot();
    });
    it("View when current path is 'admin/mastersList'", ()=>{
        const component = create(
            <MemoryRouter initialEntries={["admin/mastersList"]}>
                <LinkToEditForm {...mockProps}/>
            </MemoryRouter>
        );
        expect(component.toJSON()).toMatchSnapshot();
    });
    it("View when current path is 'admin/townsList'", ()=>{
        const component = create(
            <MemoryRouter initialEntries={["admin/townsList"]}>
                <LinkToEditForm {...mockProps}/>
            </MemoryRouter>
        );
        expect(component.toJSON()).toMatchSnapshot();
    });
    it("View when current path is 'admin/ordersList'", ()=>{
        const component = create(
            <MemoryRouter initialEntries={["admin/ordersList"]}>
                <LinkToEditForm {...mockProps}/>
            </MemoryRouter>
        );
        expect(component.toJSON()).toMatchSnapshot();
    });
});