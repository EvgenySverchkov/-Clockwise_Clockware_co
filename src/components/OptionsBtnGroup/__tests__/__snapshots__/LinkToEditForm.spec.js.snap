// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test of <LinkToEditForm /> View when current path is 'admin/clientsList' 1`] = `
<a
  className="dropdown-item"
  href="/admin/editClient/1"
  onClick={[Function]}
>
  Edit
</a>
`;

exports[`Test of <LinkToEditForm /> View when current path is 'admin/mastersList' 1`] = `
<a
  className="dropdown-item"
  href="/admin/editMaster/1"
  onClick={[Function]}
>
  Edit
</a>
`;

exports[`Test of <LinkToEditForm /> View when current path is 'admin/ordersList' 1`] = `
<a
  className="dropdown-item"
  href="/admin/editOrder/1"
  onClick={[Function]}
>
  Edit
</a>
`;

exports[`Test of <LinkToEditForm /> View when current path is 'admin/townsList' 1`] = `
<a
  className="dropdown-item"
  href="/admin/editTown/1"
  onClick={[Function]}
>
  Edit
</a>
`;
